src/series/boxBase.js

```diff
@@ -0,0 +1,128 @@
+import d3 from 'd3';
+import defined from '../util/defined';
+
+export default function() {
+
+    var xScale = d3.time.scale(),
+        yScale = d3.scale.linear(),
+        boxHigh = d3.functor(0),
+        boxLow = d3.functor(0),
+        whiskerHigh = d3.functor(0),
+        whiskerLow = d3.functor(0),
+        xValue = function(d, i) { return d.date; },
+        yValue = function(d, i) { return d.close; },
+        orient = 'vertical',
+        barWidth = d3.functor(10);
+
+    function base() { }
+
+    base.width = function(data, orientation) {
+        if (orientation === 'vertical') {
+            return barWidth(data.map(function(d, i) {
+                return xScale(xValue(d, i));
+            }));
+        } else {
+            return barWidth(data.map(function(d, i) {
+                return yScale(yValue(d, i));
+            }));
+        }
+    };
+
+    base.values = function(d, i) {
+        if (orient === 'vertical') {
+            return {
+                x: xScale(xValue(d, i)),
+                y: yScale(yValue(d, i)),
+                boxHigh: yScale(boxHigh(d, i)),
+                boxLow: yScale(boxLow(d, i)),
+                whiskerHigh: yScale(whiskerHigh(d, i)),
+                whiskerLow: yScale(whiskerLow(d, i))
+            };
+        } else {
+            return {
+                x: xScale(xValue(d, i)),
+                y: yScale(yValue(d, i)),
+                boxHigh: xScale(boxHigh(d, i)),
+                boxLow: xScale(boxLow(d, i)),
+                whiskerHigh: xScale(whiskerHigh(d, i)),
+                whiskerLow: xScale(whiskerLow(d, i))
+            };
+        }
+    };
+    base.defined = function(d, i) {
+        return defined(d, i, whiskerLow, whiskerHigh, boxLow, boxHigh, xValue, yValue);
```

Comment:
The purpose of the `defined` function is to determine whether the value accessors return suitable values for the given `d` and `i` arguments. I wouldn't expect `defined` to include the `d` and `i` arguments in its list.

See the similar implementation here:


https://github.com/ScottLogic/d3fc/blob/master/src/series/errorBase.js#L45